tests/integration/setup.coffee

```
@@ -30,38 +31,50 @@ _.assign opts,
 	retries: 3
 
-buildCredentials = ->
-	if not env
-		throw new Error('Missing environment object?!')
+exports.getSdk = getSdk
+exports.sdkOpts = opts
+exports.resin = resin = getSdk(opts)
+
+shouldDeleteTestUser = false
 
-	credentials =
-		email: env.RESINTEST_EMAIL
-		password: env.RESINTEST_PASSWORD
-		username: env.RESINTEST_USERNAME
+getCredentials = _.once Promise.method ->
+	username = "sdk_tests_#{Date.now()}"
+	register_username = "#{username}_reg"
+
+	getUserCredentials = (username) ->
+		username: username
+		email: "#{username}@resin.io"
+		password: "#{username}_!@#"
+
+	tmpEnv = env || {}
+	envTestUserCredentials =
+		username: tmpEnv.RESINTEST_USERNAME
+		email: tmpEnv.RESINTEST_EMAIL
+		password: tmpEnv.RESINTEST_PASSWORD
+
+	envPaidUserCredentials =
 		paid:
-			email: env.RESINTEST_PAID_EMAIL
-			password: env.RESINTEST_PAID_PASSWORD
-		register:
-			email: env.RESINTEST_REGISTER_EMAIL
-			password: env.RESINTEST_REGISTER_PASSWORD
-			username: env.RESINTEST_REGISTER_USERNAME
-
-	if not _.every [
-		credentials.email?
-		credentials.password?
-		credentials.username?
-		credentials.register.email?
-		credentials.register.password?
-		credentials.register.username?
-	]
-		throw new Error('Missing environment credentials')
-
-	return credentials
+			email: tmpEnv.RESINTEST_PAID_EMAIL
+			password: tmpEnv.RESINTEST_PAID_PASSWORD
 
-exports.getSdk = getSdk
+	credentials = _.assign({},
+		getUserCredentials(username),
+		envPaidUserCredentials,
+		register: getUserCredentials(register_username)
+	)
 
-exports.sdkOpts = opts
-exports.resin = resin = getSdk(opts)
+	if _.every(envTestUserCredentials)
+		console.log('Using ENV test user credentials')
+		_.assign(credentials, envTestUserCredentials)
+		return credentials
+
+	resin.auth.register
+		email: credentials.email
+		password: credentials.password
+	.tap -> shouldDeleteTestUser = true
+	.return(credentials)
+	.tapCatch ->
+		console.error('Failed to autogenerate test credentials!')
```

Perhaps this should just throw an error here?